cluster/node-pools/master-ubuntu-default/userdata.yaml

```diff
@@ -0,0 +1,571 @@
+#cloud-config
+runcmd:
+  - [ systemctl, start, gen-controller-manager-config.service ]
+  - [ systemctl, start, etcd-member.service ]
+
+write_files:
+  - owner: root:root
+    path: /etc/etcd-member/environment
+    content: |
+      ETCD_ENDPOINT={{ .Cluster.ConfigItems.etcd_endpoints }}
+
+  - owner: root:root
+    path: /etc/kubernetes/secrets.env
+    content: |
+      NODEPOOL_TAINTS=node-role.kubernetes.io/master=:NoSchedule{{if index .NodePool.ConfigItems "taints"}},{{.NodePool.ConfigItems.taints}}{{end}}
+      NODE_LABELS=node-role.kubernetes.io/master,kubernetes.io/role=master,master=true,cluster-lifecycle-controller.zalan.do/decommission-priority=999,cluster-lifecycle-controller.zalan.do/replacement-strategy=ensure-minimum-healthy-nodes,{{ .Values.node_labels }}{{if index .NodePool.ConfigItems "labels"}},{{.NodePool.ConfigItems.labels}}{{end}}
+      NODEPOOL_NAME={{ .NodePool.Name }}
+      KUBELET_ROLE=master
+
+  - owner: root:root
+    path: /var/run/s3-certs.env
+    content: |
+      S3_CERTS_BUCKET={{ .S3GeneratedFilesPath }}
+
+  - owner: root:root
+    path: /etc/kubernetes/kubeconfig
+    content: |
+      apiVersion: v1
+      kind: Config
+      clusters:
+      - name: local
+        cluster:
+          server: http://127.0.0.1:8080
+      users:
+      - name: kubelet
+      contexts:
+      - context:
+          cluster: local
+          user: kubelet
```

Comment:
Can this file be part of the ami?

Reply:
Has to be different on master and worker nodes

Reply:
Let's leave it then